crates/oxc_transformer/src/react_jsx/mod.rs

```diff
@@ -22,6 +22,11 @@ use crate::context::TransformerCtx;
 #[diagnostic(severity(warning), help("Remove `pragma` and `pragmaFrag` options."))]
 struct PragmaAndPragmaFragCannotBeSet;
 
+#[derive(Debug, Error, Diagnostic)]
+#[error("importSource cannot be set when runtime is classic.")]
+#[diagnostic(severity(warning), help("Remove `importSource` option."))]
+struct ImportSourceCannotBeSet;
```

Comment:
Shall we do a refactor in the next PR and move all the diagnostics to a separate file? e.g. https://github.com/oxc-project/oxc/blob/main/crates/oxc_parser/src/diagnostics.rs